server/routes/gitlab/gitlab_routes.py

```
@@ -0,0 +1,287 @@
+        scopes = token_info.get("scopes", [])
+
+        gitlab_token_data = {
+            "access_token": access_token,
+            "base_url": base_url,
+            "username": username,
+            "token_name": token_name,
+            "scopes": scopes,
+        }
+
+        store_tokens_in_db(user_id, gitlab_token_data, "gitlab")
+        logger.info(f"Stored GitLab credentials for user {user_id} (org-level, token_name={token_name})")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)